src/pages/admin/contacts/[id]/index.vue

```diff
@@ -121,18 +121,188 @@ meta:
 
     <div>
       <AppHeading>{{ t('contactOverview.roles') }}</AppHeading>
-      <AppInfoList v-for="role in contact.roles" :key="role.role">
-        <AppInfoListItem :name="t(`common.role.${role.role}`)">
-          <AppRoundBadge :type="isRoleCurrent(role) ? 'success' : 'danger'" />
+      <div v-for="role in contact.roles" :key="role.role">
+        <div
+          class="mt-4 flex items-center rounded border border-primary-20 px-2.5 py-1"
+        >
+          <AppRoundBadge
+            :type="isRoleCurrent(role) ? 'success' : 'danger'"
+            class="mx-2"
+          />
+          <strong class="mx-2 text-sm font-bold uppercase">
+            {{ t(`common.role.${role.role}`) }}
+          </strong>
           {{ formatLocale(role.dateAdded, 'P') + ' → ' }}
           {{
             role.dateExpires
               ? formatLocale(role.dateExpires, 'P')
               : t('contacts.data.rolesCopy.today')
           }}
-        </AppInfoListItem>
-      </AppInfoList>
+          <div class="ml-auto">
+            <AppButton variant="text" size="sm" @click="openEditRoleForm(role)"
+              >Edit</AppButton
+            >
+            <AppButton variant="dangerText" size="sm" @click="handleDeleteRole">
+              Delete role
+            </AppButton>
+          </div>
+        </div>
+        <div>
+          <AppButton
+            variant="primaryOutlined"
+            @click="isNewRoleFormVisible = true"
+            class="w-full !bg-primary-5"
+            >Add role</AppButton
+          >
+        </div>
+      </div>
+
+      <div
+        v-if="isNewRoleFormVisible"
+        class="flex flex-1 justify-center rounded border border-primary-20 py-4 px-10"
+      >
+        <form @submit.prevent="handleNewRoleFormSubmit" class="flex-initial">
```

Comment:
I think all of these forms should be broken out into separate components so they can handle their own state rather than the page needing lots of confusing variables (`isNewRoleFormVisible`, `isEditRoleFormVisible`, etc.).

Ideally the notes editor would be split out too as now there is more than 1 form on the page it would be better for them all to be separate.